hs_tracking/signals.py

```diff
@@ -1,7 +1,7 @@
 from django.contrib.auth.signals import user_logged_in, user_logged_out
 from django.dispatch import receiver
 
-from hs_core.signals import pre_download_file, post_delete_resource, post_create_resource
+from hs_core.signals import pre_download_file, pre_delete_resource, post_create_resource
```

Comment:
Just looking for a quick intuition on why here and above the logic is moving from post to pre. Are we catching something earlier this way?

Reply:
This is a workaround for hs_tracking since hs_tracking needs to use the request object to track the session and other specific request-related variables. However, moving deletion to async will lose all requests specific info, so post_delete_resource signal sent from resource deletion task will not meet hs_tracking needs. Since we don't allow to abort resource deletion task, and if a deletion is requested and put on celery query, we can just assume it will be done at least for hs_tracking purpose. There is no other listener for post resource deletion signal except for hs_tracking, hence I replaced post with pre signal for resource deletion.